gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/test/java/io/gravitee/apim/core/membership/domain_service/PrimaryOwnerDomainServiceTest.java

```diff
@@ -15,54 +15,158 @@
+    GroupQueryServiceInMemory groupQueryService = new GroupQueryServiceInMemory();
     UserCrudServiceInMemory userCrudService = new UserCrudServiceInMemory();
 
-    ApplicationPrimaryOwnerDomainService service;
+    PrimaryOwnerDomainService service;
+
+    @BeforeAll
+    static void beforeAll() {
+        UuidString.overrideGenerator(() -> "generated-id");
+        TimeProvider.overrideClock(Clock.fixed(INSTANT_NOW, ZoneId.systemDefault()));
+    }
+
+    @AfterAll
+    static void afterAll() {
+        UuidString.reset();
+        TimeProvider.overrideClock(Clock.systemDefaultZone());
+    }
 
     @BeforeEach
     void setUp() {
-        service = new ApplicationPrimaryOwnerDomainService(groupQueryService, membershipQueryService, roleQueryService, userCrudService);
+        service =
+            new PrimaryOwnerDomainService(
+                membershipCrudService,
+                roleQueryService,
+                membershipQueryService,
+                groupQueryService,
+                userCrudService
+            );
 
         roleQueryService.resetSystemRoles(ORGANIZATION_ID);
     }
 
     @AfterEach
     void tearDown() {
-        Stream.of(membershipQueryService, roleQueryService, userCrudService).forEach(InMemoryAlternative::reset);
+        Stream
+            .of(membershipCrudService, roleQueryService, membershipQueryService, groupQueryService, userCrudService)
+            .forEach(InMemoryAlternative::reset);
+    }
+
+    @Nested
+    class CreateIntegrationPrimaryOwnerMembership {
+
+        @Nested
+        class UserMode {
+
+            @Test
+            void should_create_an_user_api_primary_owner_membership() {
+                // When
+                service.createIntegrationPrimaryOwnerMembership(
+                    INTEGRATION_ID,
+                    PrimaryOwnerEntity.builder().id(MEMBER_ID).type(PrimaryOwnerEntity.Type.USER).build(),
+                    AUDIT_INFO
+                );
+
+                // Then
+                assertThat(membershipCrudService.storage())
+                    .containsExactly(
+                        Membership
+                            .builder()
+                            .id("generated-id")
+                            .roleId(integrationPrimaryOwnerRoleId(ORGANIZATION_ID))
+                            .memberId(MEMBER_ID)
+                            .memberType(Membership.Type.USER)
+                            .referenceId(INTEGRATION_ID)
+                            .referenceType(Membership.ReferenceType.INTEGRATION)
+                            .source("system")
+                            .createdAt(INSTANT_NOW.atZone(ZoneId.systemDefault()))
+                            .updatedAt(INSTANT_NOW.atZone(ZoneId.systemDefault()))
+                            .build()
+                    );
+            }
+        }
+
+        @Nested
+        class GroupMode {
+
+            @Test
+            void should_create_a_group_api_primary_owner_membership() {
```

Comment:
Based on the last week exchange, it has been decided to not support Group PrimaryOwner for Integration